posix_case_conversion/ext/posix_case_conversion/posix_case_conversion.c

```diff
@@ -24,11 +24,30 @@ static rb_encoding *utf8_enc = NULL;
 #define MAX_ENC_LEN 4
 
 /*
- * Downcases the supplied string according to POSIX locale semantics. Returns
- * a freshly allocated string that the caller is responsible for.
+ * Downcases the supplied string according to US-ASCII semantics.
+ * Returns a freshly allocated string that the caller is responsible for.
  */
 static char *
-posix_downcase(const char *string, size_t len)
+posix_downcase_usascii(const char *string, size_t len)
+{
+  char *lower_str;
+  size_t curr_char;
+  int c;
+  lower_str = (char *) malloc(len);
+  for (curr_char = 0; curr_char < len; curr_char++)
+    {
+      c = string[curr_char];
+      lower_str[curr_char] = (c >= 'A' && c <= 'Z') ? c + ('a' - 'A') : c;
```

Comment:
@pritambaral: Although we discussed this, I was wondering why not use `tolower` instead which I assume does the same thing?

Reply:
`tolower` is not the same as this, but the difference matters only for non-ASCII characters.